File: jv-service/src/main/java/com/lcaohoanq/jvservice/domain/subcategory/Subcategory.java
package com.lcaohoanq.jvservice.domain.subcategory;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.lcaohoanq.jvservice.base.entity.BaseEntity;
import com.lcaohoanq.jvservice.domain.category.Category;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@Entity
        @Table(name = "subcategories")
@JsonPropertyOrder({
    "id",
    "name",
    "category"
})
public class Subcategory extends BaseEntity implements Comparable<Subcategory> {

    @Id
    @SequenceGenerator(name = "subcategories_seq", sequenceName = "subcategories_id_seq", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "subcategories_seq")
    @Column(name="id", unique=true, nullable=false)
    @JsonProperty("id")
    private Long id;
    private String name;
    
    @ManyToOne
    @JoinColumn(name = "category_id")
    @JsonIgnore
    private Category category;

    @Override
    public int compareTo(Subcategory that) {
        return this.id.compareTo(that.id);
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }
}

File: kt-service/src/main/kotlin/com/lcaohoanq/ktservice/entities/SubCategory.kt
package com.lcaohoanq.ktservice.entities

import BaseEntity
import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "subcategories")
class SubCategory: BaseEntity(), Comparable<SubCategory> {

    @Id
    val id: Long? = null
    val name: String? = null

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    @JsonIgnore
    val category: Category = Category()

    override fun compareTo(other: SubCategory): Int {
        return compareValues(this.id, other.id)
    }

}